checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -82,20 +90,26 @@
 public class MustCallInference {
 
   /**
-   * The fields that have been inferred to be disposed within the CFG (i.e., method) currently under
-   * analysis. All of the fields in this set when inference finishes will be given an @Owning
-   * annotation. Note that this set is not monotonically-increasing: fields may be added to this set
-   * and then removed during inference. For example, if a field's must-call method is called, it is
-   * added to this set. If, in a later statement in the same method, the same field is re-assigned,
-   * it will be removed from this set (since the previously-inferred closing of the obligation is
-   * invalid).
+   * The fields that have been inferred to be disposed within the current method. When inference
+   * finishes, all of the fields in this set will be given an @Owning annotation. Note that this set
+   * is not monotonically-increasing: fields may be added to this set and then removed during
+   * inference. For example, if a field's must-call method is called, it is added to this set. If,
+   * in a later statement in the same method, the same field is re-assigned, it will be removed from
+   * this set (since the field assignment invalidated the previously-inferred closing of the
+   * obligation).
    */
   private final Set<VariableElement> disposedFields = new HashSet<>();
```

Comment:
This might cause nondeterministic ordering of fields in `@EnsuresCalledMethods` annotations, and to prevent this, I've switched to using `LinkedHashSet`.